ambry-messageformat/src/main/java/com.github.ambry.messageformat/MessageFormatRecord.java

```diff
@@ -468,7 +472,66 @@ public static boolean deserializeDeleteRecord(CrcInputStream crcStream) throws I
             "corrupt data while parsing delete record Expected CRC " + expectedCRC + " Actual CRC " + actualCRC);
         throw new MessageFormatException("delete record data is corrupt", MessageFormatErrorCodes.Data_Corrupt);
       }
-      return isDeleted;
+      return new DeleteRecord();
+    }
+  }
+
+  /**
+   *  - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+   * |         |               |               |               |             |
+   * | version |   AccountId   |  ContainerId  |  DeletionTime |     Crc     |
+   * |(2 bytes)|    (2 byte2)  |   (2 bytes)   |   (4 bytes)   |  (8 bytes)  |
+   * |         |               |               |               |             |
+   *  - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+   *  version         - The version of the delete record
+   *
+   *  AccountId     - AccountId that the blob belongs to
+   *
+   *  ContainerId   - ContainerId that the blob belongs to
+   *
+   *  Deletion Time - Time of deletion in secs
+   *
+   *  Crc             - The crc of the delete record
+   *
+   */
+  public static class Delete_Format_V2 {
+
+    public static final int AccountId_Field_Size_In_Bytes = 2;
```

Comment:
I would prefer having an underscore before Id b/c we did that for `Deletion_Time...`